lib/hal/portable/st/aws_hal_perf_cntr.c

```diff
@@ -0,0 +1,233 @@
+/*
+* Amazon FreeRTOS V1.x.x
+* Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*
+* http://aws.amazon.com/freertos
+* http://www.FreeRTOS.org
+*/
+
+/**
+* @file aws_hal_perf_cntr.c
+* @brief File of the implementation for performance counter APIs calling STM drivers.
+*/
+
+#include <stdint.h>
+
+/* ST HAL APIs to be called in this file. */
+#include "stm32l4xx_hal.h"
```

Comment:
should this header be included stm32l4xx_hal_tim.h?

Reply:
I actually tried stm32l4xx_hal_tim.h first, which failed to compile. (That was when I was trying to implement error handling path for _open(), and HAL_StatusTypeDef was missing from *_tim.h header.) I'll meditate over this. If all we have to do is ignoring errors, then *_tim.h would do the work.